app/assets/javascripts/pages/agent-edit-page.js.coffee

```diff
@@ -108,10 +108,12 @@ class @AgentEditPage
     $(".control-link-region").show()
 
   hideEventCreation: ->
-    $(".event-related-region").hide()
```

Comment:
Wow! True, that bug existed before for `sources` and `control targets`. 998af238f9da05fb44738cda5c1542a67369791c prevents that from happening but also made the `AgentEditPage` even more complex then it was before.
If the performance impact would not be so bad, I would argue for re-rending the form every time the agent type was changed as this would allow us to remove a lot of the javascript code.

I will open a separate pull request with a migration that removes all the useless agent connections.